File: nodejs-webdriverio/features/step_definitions/search-google.js
module.exports = function () {

    var targetLink;

    this.Given(/^I am on the Google homepage$/, function (callback) {
        browser.url('http://google.co.uk').call(callback);
    });

    this.Then(/^I will search for "(.*)"$/, function (searchText, callback) {
        browser
            .setValue('#lst-ib', searchText)
            .keys('Enter')
            .pause(1000)
            .waitForExist('body')
            .call(callback);
    });

    this.Then(/^I should see "(.*)"$/, function (expectedText, callback) {
        targetLink = expectedText;
        browser
            .waitForText('//h3/a[text() = "' + targetLink + '"]')
            .call(callback);
    });

    this.Then(/^I will click the (.*) link$/, function (linkName, callback) {
        browser
            .click('//h3/a[text() = "' + targetLink + '"]/../../..//h3/a[text() = "' + linkName + '"]')
            .call(callback);
    });

};

File: ruby-selenium/features/step_definitions/search-google.rb

Given(/^I am on the Google homepage$/) do
  visit 'http://www.google.co.uk'
end

Then(/^I will search for "([^"]*)"$/) do |searchText|
  fill_in 'lst-ib', :with => searchText
end

Then(/^I should see "([^"]*)"$/) do |expectedText|
  expect(page).to have_content(expectedText)
end

Then(/^I will click the Games link$/) do
  click_link('Games')
end
